agent/taskresource/asmauth/asmauth.go

```diff
@@ -293,6 +295,16 @@ func (auth *ASMAuthResource) retrieveASMDockerAuthData(asmAuthData *containerres
 	return nil
 }
 
+func (auth *ASMAuthResource) getIAMCredentials() (credentials.IAMRoleCredentials, bool, string) {
+	if auth.isContainerCredentials {
+		executionCredentials, ok := auth.credentialsManager.GetContainerCredentials(auth.GetExecutionCredentialsID())
```

Comment:
hmm.. now I am wondering: if the credentials are retrieved by some random `id`, why is it necessary to separate by `credentialsManager.GetContainerCredentials` and `credentialsManager.GetTaskCredentials`? in other words, why can't the credentials manager by agnostic as to the credentials provenance (task vs container)?

Reply:
The reason for having task vs container was just for semantics purpose. 
Retrieving container credentials from a GetTaskCredentials() would be kind of confusing:(

Reply:
I was thinking more to just have a single methof `credentialsManager.GetCredentials`. Not sure why the credentials manager itself has to be aware of container vs task credentials if all that matters to retrieve is the unique ID.

Reply:
Hmmm I kind of prefer storing task vs container credentials in separate maps as it would be helpful in future to use these flexibly but I can't think a particular use case as of now!

Reply:
not feeling super strong about this, so be it :)